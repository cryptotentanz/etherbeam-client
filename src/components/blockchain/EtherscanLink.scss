@use '../../styles/resources/sizes';

.my-etherscanlink-img {
  display: inline-block;
  background-color: #ffffff;
  border-radius: 50%;
  vertical-align: middle;
  padding: 1px;
  width: sizes.$size;
  height: sizes.$size;

  img {
    width: 100%;
    height: 100%;
    margin: auto;
  }
}
