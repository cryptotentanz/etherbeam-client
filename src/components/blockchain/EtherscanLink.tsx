import React, { FC } from 'react'
import Link from '../Link'
import './EtherscanLink.scss'

type SizeProp = 'm' | 'l' | 'xl'

interface EtherscanLinkProps {
  url: string
  size?: SizeProp
}

const EtherscanLink: FC<EtherscanLinkProps> = ({ url, size }: EtherscanLinkProps) => {
  return (
    <Link href={url} description="Etherscan" external className={size ? `my-size-${size}` : ''}>
      <div className="my-etherscanlink-img">
        <img src="/images/etherscan.svg" />
      </div>
    </Link>
  )
}

export default EtherscanLink
