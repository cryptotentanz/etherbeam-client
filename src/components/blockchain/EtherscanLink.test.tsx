import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import EtherscanLink from './EtherscanLink'

describe('<EtherscanLink />', () => {
  const url = 'https://etherscan.io/address/0x0000000000000000000000000000000000000111'
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<EtherscanLink url={url} />)
    })

    it('should render', () => {
      const link = component.find('Link')
      expect(link.prop('href')).toBe(url)
      expect(link.prop('description')).toBe('Etherscan')
      expect(link.prop('external')).toBeTruthy()

      const div = link.find('div')
      expect(div.hasClass('my-etherscanlink-img')).toBeTruthy()
      expect(div.find('img').at(0).exists()).toBeTruthy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Size', () => {
    beforeAll(() => {
      component = shallow(<EtherscanLink url={url} size="l" />)
    })

    it('should render', () => {
      const link = component.find('Link')
      expect(link.hasClass('my-size-l')).toBeTruthy()
    })
  })
})
