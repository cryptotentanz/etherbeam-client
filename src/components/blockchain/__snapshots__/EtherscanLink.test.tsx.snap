// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EtherscanLink /> Default should match snapshot 1`] = `
<a
  class="my-clickable  my-ulink my-secondary-fg"
  href="https://etherscan.io/address/0x0000000000000000000000000000000000000111"
  rel="noreferrer"
  tabindex="0"
  target="_blank"
  title="Etherscan"
>
  <div
    class="my-etherscanlink-img"
  >
    <img
      src="/images/etherscan.svg"
    />
  </div>
</a>
`;
